File: JAVASCRIPT/Algorithms/insertionSort.js
const insertionSort = (arr) => {
  for (let i = 1; i < arr.length; i++) {
    let currentValue = arr[i];
    let j;
    for (j = i - 1; j >= 0 && arr[j] > currentValue; j--) {
      arr[j + 1] = arr[j];
    }
    arr[j + 1] = currentValue;
  }
  return arr;
};

// console.log(insertionSort([45, 3, 79, 14, 459, 21, 999])); // Output: [3, 14, 21, 45, 79, 459, 999];

File: JAVA/DSA in java/sorting/insertionsort.java
public class insertionsort {
    public static void printarr(int arr[]){
        for(int i=0;i<arr.length;i++){
            System.out.print(arr[i]+" ");
        }
        System.out.println();
    }
    
public static void insertion(int arr[]){
    for(int i=1;i<arr.length;i++){
        int curr=arr[i];
        int prev=i-1;
        while(prev>=0 && arr[prev]>curr){
            arr[prev+1]=arr[prev];
            prev--;
        }
        //insertion
        arr[prev+1]=curr;
     }
}
    public static void main(String args[]){
    int arr[] = {1,4,1,3,2,4,3,7};
    insertion(arr);
    printarr(arr);
    }
}
